public/app/features/library-panels/components/PanelLibraryOptionsGroup/PanelLibraryOptionsGroup.tsx

```diff
@@ -55,8 +55,12 @@ export const PanelLibraryOptionsGroup: FC<Props> = ({ panel, searchQuery }) => {
 
     panel.configRev = 0;
     panel.refresh();
-    panel.events.publish(new PanelQueriesChangedEvent());
-    panel.events.publish(new PanelOptionsChangedEvent());
+    const unsubscribeEvent = panel.events.subscribe(PanelDirectiveReadyEvent, () => {
+      panel.refresh();
```

Comment:
Maybe we should remove `panel.refresh();` on line 57?

Reply:
Aha, ok I understand thank you for clarifying it. Maybe we should add a comment about this or add it to React panels too as you already suggested 👍